src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingAggregatedLogger.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Telemetry;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class LinkedFileDiffMergingAggregatedLogger : IAggregatedTelemetryLogger
+    {
+        private static object globalLock = new object();
+        private static Dictionary<Workspace, LinkedFileDiffMergingAggregatedLogger> loggers = new Dictionary<Workspace, LinkedFileDiffMergingAggregatedLogger>();
```

Comment:
there is no one cleaning this up. so ConditionalWeakTable might be better? especially cases such as preview workspace and others? unless you specifically do checking to handle those cases?